usr/src/cmd/cmd-inet/usr.sbin/ipadm/ipadm.c

```diff
@@ -1643,8 +1643,7 @@ is_from_gz(const char *lifname)
 
 	if (if_info->ifi_cflags & IFIF_L3PROTECT)
 		ret = _B_TRUE;
-	if (if_info)
-		ipadm_free_if_info(if_info);
+	ipadm_free_if_info(if_info);
```

Comment:
hmmm I thought we had this already ...

Reply:
This is one of the things that the new `smatch` (lint replacement) is throwing up. The check is redundant (for one thing, if_info has already been de-referenced just above) and smatch complains. jlevon is working on getting the gate to be smatch clean, then smatch can be globally enabled as an additional shadow compiler.